identity/webapp/src/routes/api/get-user.ts

```diff
@@ -0,0 +1,12 @@
+import { RouteMiddleware } from '../../types/application';
+import { callRemoteApi } from '../../utility/api-caller';
+// This will communicate with the backend api to make the call on our behalf.
+
+export const getUser: RouteMiddleware = async (context) => {
+  const idNumber = context.state.user.profile.id.replace('auth0|p', '');
```

Comment:
This feels grotty. Is there a better way to get the user ID in a form that the API expects?

`context.state.user.profile.id` is something like `"auth0|p123456"` but the API only expects the `123456` numeric part.

Reply:
I was thinking for this we could look at storing the `123456` somewhere on the session when we get a successful login response. That way we only have to do this once per session rather than each time we need it to call the API (which I think is every time we call the API from the account management app). Not quite sure what the best way of storing something in the session is, but I think the callback in the `Auth0Strategy` in `configure-auth0.ts` is probably the place to do it.

Reply:
Great shout. Putting it in the `deserializeUser` method seems to work.

https://stackoverflow.com/questions/19313332/adding-extra-information-in-user-object-in-passport-js-or-somewhere-in-session